app.js

```diff
@@ -81,28 +90,60 @@ var imageConstructorArray = [
   shark,
   sweep,
   tauntaun,
-  thumbs,
   unicorn,
   usb,
   waterCan,
   wineGlass,
-] ;
+];
+var percentArray = [];
+var threePic = [];
+var oldPics = [];
+var numOfPics = 3;
 // populates three images pulled at an index from the array. the index is a random number
 function populateThree(){
-  for (var i = 0; i < 3; i++) {
+  oldPics = threePic;
+  threePic = [];
+  while (threePic.length < numOfPics) {
     var integerIndex = getRandomInt();
-    //this makes the number we generated storeable since calling the functon just makes a new number
-    imageConstructorArray[integerIndex].addPicture(integerIndex);
+    if (threePic.indexOf(integerIndex) === -1 && oldPics.indexOf(integerIndex) === -1) {
+      threePic.push(integerIndex);
+    }
   }
+  console.log('threePic: ', threePic);
+  console.log('oldPics: ', oldPics);
 };
+function showThree(){
+  for(var j = 0; j < threePic.length; j++){
+    imageConstructorArray[threePic[j]].addPicture(threePic[j]);
+  }
+}
+
 //calling the function
 populateThree();
+showThree();
 
 //handle click event logs the click and once it logs and wipes it populates three more
 function handleClick(event){
-  console.log('handleClick', event);
-  console.log(event.target.dataset.integerIndex);
-  console.log(imageConstructorArray[event.target.dataset.integerIndex]);
-  logClick(imageConstructorArray[event.target.dataset.integerIndex]);
-  populateThree();
+  if(totalClicks < clickLimit){
+    console.log('handleClick', event);
```

Comment:
This log is probably there to show you when the handleClick function runs. But, since you've nested it in an if statement, it only does that some of the time. If you bump it up to the first line in the function, it'll do it's job better.